pkg/util/clusterauthorizer/authorizer.go

```diff
@@ -44,36 +48,46 @@ func NewAzRefreshableAuthorizer(log *logrus.Entry, azEnv *azureclient.AROEnviron
 		log:              log,
 		azureEnvironment: azEnv,
 		kubernetescli:    kubernetescli,
-		tokenClient:      tokenClient,
 	}, nil
 }
 
-func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Context) (refreshable.Authorizer, error) {
+func (a *azRefreshableAuthorizer) NewRefreshableAuthorizerToken(ctx context.Context) (autorest.Authorizer, error) {
 	// Grab azure-credentials from secret
 	credentials, err := AzCredentials(ctx, a.kubernetescli)
 	if err != nil {
 		return nil, err
 	}
-	// create service principal token from azure-credentials
-	token, err := a.tokenClient.GetToken(ctx,
-		a.log,
-		string(credentials.ClientID),
-		string(credentials.ClientSecret),
-		string(credentials.TenantID),
-		a.azureEnvironment.ActiveDirectoryEndpoint,
-		a.azureEnvironment.ResourceManagerEndpoint)
+
+	tokenCredential := a.tokenCredential // for unit testing only
+	if tokenCredential == nil {
+		// Create service principal token from azure-credentials
+		tokenCredential, err = azidentity.NewClientSecretCredential(
+			string(credentials.TenantID),
+			string(credentials.ClientID),
+			string(credentials.ClientSecret),
+			a.azureEnvironment.ClientSecretCredentialOptions())
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	scopes := []string{a.azureEnvironment.ResourceManagerEndpoint + "/.default"}
+
+	tokenRequestOptions := policy.TokenRequestOptions{Scopes: scopes}
+	token, err := tokenCredential.GetToken(ctx, tokenRequestOptions)
 	if err != nil {
 		return nil, err
 	}
 
+	// XXX Is this validation still necessary?
```

Comment:
todo?

Reply:
Was originally an on-the-fly test to make sure the provided ADAL token was valid, or at least parsable.  Unclear why it was needed, [commit that introduced](https://github.com/Azure/ARO-RP/commit/144606bb004b64623d4eec969c7098253f7ba56a#diff-9b05764fc49f59be0df3e53cc7ff53930ad08391ac5a4e3740f70960db70a73d) it doesn't explain.  Hence my comment.

Wrote that early on when I was still figuring things out.  I'd lean now toward deleting this part.  I trust no `GetToken` implementation in azidentity is going to return an invalid token.